clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -409,4 +413,13 @@ protected ApplicationEventProcessor create() {
             }
         };
     }
+
+    private void process(final SeekUnvalidatedEvent event) {
+        try {
+            subscriptions.seekUnvalidated(event.partition(), event.position());
```

Comment:
If we update the event class, we can then move more of the logic here:

```suggestion
            SubscriptionState.FetchPosition newPosition = new SubscriptionState.FetchPosition(
                event.offset(),
                event.offsetEpoch(),
                metadata.currentLeader(event.partition())
            );
            subscriptions.seekUnvalidated(event.partition(), newPosition);
```